runners/core-java/src/main/java/org/apache/beam/runners/core/DoFnRunners.java

```diff
@@ -116,4 +116,24 @@
         stepContext.timerInternals(),
         droppedDueToLatenessAggregator);
   }
+
+  /**
+   * Returns an implementation of {@link DoFnRunner} that handles
+   * late data dropping and garbage collect for stateParDo.
```

Comment:
I think the last bit would look better as `"and garbage collection for stateful {@link DoFn DoFns}."`